jxl/src/render/stages/save.rs

```diff
@@ -12,19 +12,25 @@ use crate::{
 pub struct SaveStage<T: ImageDataType> {
     buf: Image<T>,
     channel: usize,
+    scale: T,
```

Comment:
I am not sure this is the best thing to do (intuitively, I think we should probably have a fixed scale for each data type, and have datatype conversions do the scaling), but I think it might be OK for now.